### PR TITLE
refactor: use method signature for class overloads

### PR DESCRIPTION
**Pull request by ifiokjr**
## Description

Switch to using method signatures for extension class methods as discussed in #360.

## Checklist

- [x] I have read the [**contributing**](https://github.com/remirror/remirror/blob/master/docs/contributing.md) document.
- [x] My code follows the code style of this project and `pnpm fix` completed successfully.
- [x] I have updated the documentation where necessary.
- [x] New code is unit tested and all current tests pass when running `pnpm test`.

### CODE REVIEW

**Comment by ifiokjr** [comment]
@ocavue I won't make any further changes so feel free to take over.

File: packages/@remirror/core/src/manager/remirror-manager.ts [hunk 1]
```diff
@@ -347,19 +339,19 @@ export class RemirrorManager<Combined extends AnyCombinedUnion> {
       const destroyHandler = extension.onDestroy;
 
       if (createHandler) {
-        this.#handlers.create.push(createHandler);
+        this.#handlers.create.push(createHandler.bind(extension));
```
Comment (ocavue) [inline packages/@remirror/core/src/manager/remirror-manager.ts:342]: @ifiokjr You may need to check this modification.

**Comment by ocavue** [comment]
@ifiokjr 

I've updated all four lifecycle methods to use the method signature. Combine with your modification for methods inside `ExtensionCreatorMethods`, it seems that all methods have been updated.

File: packages/@remirror/core/src/manager/remirror-manager.ts (hunk 1 above)
Comment (ifiokjr) [reply packages/@remirror/core/src/manager/remirror-manager.ts:342]: This looks good to me. Are there any potential pitfalls?

**Comment by ocavue** [comment]
> I was thinking of taking advantage of this PR to update all the extensions which are currently using class properties to defined their instance methods.

ok, I can do that.

File: .changeset/cool-countries-pay.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+'@remirror/core': minor
+'@remirror/extension-bidi': patch
+'@remirror/extension-placeholder': patch
+---
+
+Switch to using method signatures for extension class methods as discussed in #360.
+
+Remove the first parameter `extenisons` from method `onCreate`, `onView` and `onDestroy`.
```
Comment (ifiokjr) [inline .changeset/cool-countries-pay.md:9]:
```suggestion
Remove the first parameter `extensions` from method `onCreate`, `onView` and `onDestroy`.
```

File: .changeset/cool-countries-pay.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+'@remirror/core': minor
```
Comment (ifiokjr) [inline .changeset/cool-countries-pay.md:2]: Would be awesome if you could also add every extension that is affected by the changes.

File: .changeset/cool-countries-pay.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+'@remirror/extension-bidi': patch
+'@remirror/extension-placeholder': patch
+---
+
+Remove the first parameter `extensions` from method `onCreate`, `onView` and `onDestroy`.
```
Comment (ifiokjr) [inline .changeset/cool-countries-pay.md:7]:
```suggestion
Remove the first parameter `extensions` from the lifecycle methods `onCreate`, `onView` and `onDestroy`.
```

File: .changeset/cool-countries-pay.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+Switch to using method signatures for extension class methods as discussed in #360. The follow methods have been affected:
```
Comment (ifiokjr) [inline .changeset/cool-countries-pay.md:9]:
```suggestion
Switch to using method signatures for extension class methods as discussed in #360. The following methods have been affected:
```

**Comment by ifiokjr** [comment]
It's great to see the progress you're making with this. You probably know the codebase better than me at this point 😄

**Comment by ocavue** [comment]
> You probably know the codebase better than me at this point

Thanks for your compliment. I just used some regex expressions to do the job and I still have a lot to learn about the codebase.

**Pull request merged**